tests/trogdor/trogdor_runner.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import json
+import sys
+from produce_spec_runner import execute_produce_spec
+from trogdor_utils import update_trogdor_error
+from trogdor_utils import trogdor_log
+
+parser = argparse.ArgumentParser(description='Python Trogdor Producer.')
+parser.add_argument('--spec', dest='spec', required=False)
+args = parser.parse_args()
+
+
+def execute_task(workload):
+    try:
+        spec_class = workload["class"]
+        if spec_class == 'org.apache.kafka.trogdor.workload.ProduceBenchSpec':
+            execute_produce_spec(workload)
+        else:
+            update_trogdor_error(spec_class + " is not supported.")
+    except Exception as exp:
+        update_trogdor_error("Exception: " + exp)
+
+
+def get_task_from_specfile(specFile):
+    """
+    :param specFile: One Trogdor ExternalCommandSpec JSON file
+     One example of the ExternalCommandSpec spec:
+
+    :return specFile["spec"]["workload"] or None
+    """
+    specString = ""
+    workload = None
+    with open(specFile) as f:
+        for line in f:
+            if line.startswith('#'):
+                continue
+            specString += line
+        f.close()
+    if (specString == ""):
+        return None
+    task = json.loads(specString)
+    if "class" in task and "workload" in task and \
+            task["class"] == "org.apache.kafka.trogdor.workload.ExternalCommandSpec":
+        workload = task["workload"]
+    return workload
+
+
+def get_task_from_input():
+    """
+    Waiting for a new task on stdin, formatted in {"id":<task ID string>, "workload":<configured workload JSON object>}
+    :return: spec["workload"]
+    """
+    for line in sys.stdin:
+        try:
+            comm = json.loads(line)
+            if "workload" in comm and "id" in comm:
+                trogdor_log("Received one workload:{}".format(comm))
+                return comm["workload"]
+        except ValueError:
+            trogdor_log("Input line " + line + " is not a valid external runner command.")
+
+
+if __name__ == '__main__':
+    spec = None
```

Comment:
spec can't be undefined after the if statement, so no need to initialize here.

Reply:
afaict it's the nicest way to do it though? `spec` needs to existing in this scope, and this seems the most straightforward way to this.